### ISSUE TITLE
Can't plot route to Nenia

### ISSUE THREAD

**Issue by McloughlinGuy** [endless-sky/endless-sky#7865]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

With the current save I'm on, if you click Nenia, It won't plot a route to it. If you try to follow the route anyways, it will treat it like you're trying to jump to a system you can't reach without a jump drive. If you click on the system you're in and then try it again, it seems to work fine. Still, this is really freaking weird.

### Steps to Reproduce

1. Use the save file provide
2. Click on Nenia on the map (Do NOT click on any other systems beforehand)
3. See that it won't plot a route to it

### Expected Behavior

It should let me make a route to the system when I click on Nenia.

### Screenshots

![Screenshot (218)](https://user-images.githubusercontent.com/70719891/207183240-d18db575-3149-4f26-a440-219489bc29cd.png)

https://user-images.githubusercontent.com/70719891/207183445-6a78870f-8d11-40a1-834c-71eee46ef6ef.mp4

https://user-images.githubusercontent.com/70719891/207183453-b6e50f92-a065-4a36-9027-f363eeb24f7a.mp4

### Link to save file

[Xavier Illini.txt](https://github.com/endless-sky/endless-sky/files/10212889/Xavier.Illini.txt)

### Operating System

Windows 10

### Game Source

ESLauncher2

### Game Version

Continuous

### Additional Information

I think this happened because I switched my jump enabled Leviathan to my non jump enabled Puffin.

**Comment by warp-core** [comment]
I suspect your initial route was plotted while the "Thunder Pearl" was your flagship. That ship has a jump drive and so the route would involve jumping from Peragenor directly to Segesta. When you switched to "Shadow in the Depths", this route was no longer possible, as that ship lacks a jump drive. The game does not attempt to update the travel plan when you click on the current destination again, however selecting any other system (including any system along the existing route) should cause it to update and this is why, after selecting your current system and reselecting Nenia, the route is recalculated and accounts for the lack of jump drive on your current flagship.

**Comment by warp-core** [comment]
> > I suspect your initial route was plotted while the "Thunder Pearl" was your flagship. That ship has a jump drive and so the route would involve jumping from Peragenor directly to Segesta. When you switched to "Shadow in the Depths", this route was no longer possible, as that ship lacks a jump drive. The game does not attempt to update the travel plan when you click on the current destination again, however selecting any other system (including any system along the existing route) should cause it to update and this is why, after selecting your current system and reselecting Nenia, the route is recalculated and accounts for the lack of jump drive on your current flagship.
> 
> Oh I see. Maybe we should make the game attempt to update the travel plan when you click the destination again, but it might not be worth doing for such a niche issue.

I don't think that would be unreasonable. The game has already established that the existing travel plan is incompatible with the flagship: that's why it's not displaying the whole thing in the map.

**Comment by Amazinite** [comment]
If I load the save file then I can see the behaviors that's described, but if I try to reproduce it by starting the travel plan with the Leviathan then switch to the Puffin, I'm able to click on Nenia and get an updated travel plan.

My guess is this was fixed back in 0.10.3 by https://github.com/endless-sky/endless-sky/commit/9eb74d26b3570b74382ca88d61c684682423a805.
